test/additionalTests/LP9test/testLP9.m

```diff
@@ -0,0 +1,87 @@
+% The COBRAToolbox: Lp9.m
+%
+% Purpose:
+%     - test if bug fix in LP9 affects the results for test model (without
+%     coupling constraints)
+%
+% Authors:
+%     - fix compatibility for models with coupling constraints: Agnieszka
+%     Wegrzyn 2016/03/06
+%
+
+global CBTDIR
+
+% define the features required to run the test
+requiredToolboxes = {};
+
+requiredSolvers = {'glpk'};
+
+% require the specified toolboxes and solvers, along with a UNIX OS
+solversPkgs = prepareTest('requiredSolvers', requiredSolvers, 'requiredToolboxes', requiredToolboxes);
+
+% save the current path and initialize the test
+currentDir = cd(fileparts(which(mfilename)));
+
+% determine the test path for references
+testPath = pwd;
+
+% set the tolerance
+tol = 1e-8;
+
+% load the model
+model = getDistributedModel('ecoli_core_model.mat'); %For all models in the test/models folder and subfolders
+
+% define core reactions as in: extractionTranscriptomics tutorial
+core = [1;3;7;10;18;42;43;49;50;53;55;57;72;75];
+epsilon = 1e-4;
+singleton = false;
+[nMets,nRxns] = size(model.S);
+
+for k = 1:length(solversPkgs.LP)
+    fprintf(' -- Running <testFile> using the solver interface: %s ... ', solversPkgs.LP{k});
+
+    solverLPOK = changeCobraSolver(solversPkgs.LP{k}, 'LP', 0);
+
+    if solverLPOK
+        % define input variables
+        % from fastcore.m:
+        LPproblem = buildLPproblemFromModel(model);
+        % reactions irreversible in the reverse direction
+        Ir = find(model.ub<=0);
+        % flip direction of reactions irreversible in the reverse direction
+        LPproblem.A(:,Ir) = -LPproblem.A(:,Ir);
+        tmp = LPproblem.ub(Ir);
+        LPproblem.ub(Ir) = -LPproblem.lb(Ir);
+        LPproblem.lb(Ir) = -tmp;
+        % Find irreversible reactions
+        irrevRxns = find(model.lb>=0);
+        % Find irreversible core reactions
+        J = intersect(core, irrevRxns);
+        %Find all the reactions that are not in the core
+        nbRxns = 1:nRxns;
+        % Non Core reactions (penalized)
+        P = setdiff(nbRxns, core);
+        % from findSparseMode.m:
+        basis=[];
+        if singleton
+          [V, basis] = LP7(J(1), model, LPproblem, epsilon, basis);
+        else
+          [V, basis] = LP7(J, model, LPproblem, epsilon, basis);
+        end
+        K = intersect(J, find(V >= 0.99*epsilon));
+        % create reference data
+        V_ref = LP9(K, P, model, LPproblem, epsilon);
```

Comment:
`V_ref` should be loaded from a `.mat` file. Otherwise, we will have 2 versions of the code, one version in `src`, and one version in `test` (and named differently).